src/procutils.c

```diff
@@ -89,3 +94,43 @@ int o_redirect(int mode, const char *file)
 
 	return has_err;
 }
+
+int pid_of(const char* exe_name)
+{
+    DIR *dir = opendir("/proc");
+    if (dir == NULL)
+        return 0;
+    struct dirent *ent;
+    while ((ent = readdir(dir)) != NULL)
+    {
+        int pid = 0;
+        const char *ptr = ent->d_name;
+        if (ptr == NULL)
+            continue;
+
+        for (; *ptr && isdigit(*ptr); ptr++) // Convert string to integer
+            pid = pid * 10 + (*ptr - '0');
+        if (*ptr != '\0')
+            continue;
+
+        char cmdline_file[(ptr - ent->d_name) + 16];
+        snprintf(cmdline_file, sizeof(cmdline_file), "/proc/%s/cmdline", ent->d_name);
+
+        FILE *file = fopen(cmdline_file, "r");
+        if (file == NULL)
+            continue;
+
+        char name[200] = { 0 };
```

Comment:
You can reuse the same buffer for here too. call it some thing generic like "buf".